src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1682,13 +1704,16 @@ LedgerManagerImpl::applyLedger(LedgerCloseData const& ledgerData,
             appliedLedgerState->getSorobanConfig());
     }
 
+    // Step 5
+    maybeRunSnapshotInvariants(appliedLedgerState);
```

Comment:
Any reason not to have the invariant manager initiate the check itself? This would help with separation of responsibilities - right now it's a bit weird that LedgerManager knows about internal details, like triggering it periodically etc. I think simplest would be to have a timer in the invariant and periodically trigger the check by asking LM for the latest LCL state. Note that the config would also be in seconds, not in ledgers - given our current plans to start changin block times, I think it's generally safer to reason in absolute time, and not ledger counts.

Reply:
I disagree for a few reasons. First, the state we're checking here is not public outside of the LedgerManager. If we were just relying on bucket snapshots, I agree that asking the LM for LCL state is fine. However, since we need to make a special copy of internal LedgerManager state that's not thread safe (the soroban caches), I think this interface would be much more complex. If we went to wall-clock time instead of ledgers, or if we asked the LM for latest state on the non-apply thread, it would be much more complex, as we'd have to "wait" until the apply manager was in a safe state to make the copy.